rust/connlib/tunnel/src/tests/sut.rs

```diff
@@ -98,25 +99,33 @@ impl StateMachineTest for TunnelTest {
             debug_span!("gateway"),
         );
 
-        let relay = ref_state.relay.map(
-            |relay, ip4, ip6| {
-                relay.map(|seed| {
-                    firezone_relay::Server::new(
-                        IpStack::from((ip4, ip6)),
-                        rand::rngs::StdRng::seed_from_u64(seed),
-                        3478,
-                        49152,
-                        65535,
-                    )
-                })
-            },
-            debug_span!("relay"),
-        );
-        let portal = SimPortal::new(client.inner().id, gateway.inner().id, relay.inner().id);
+        let relays = ref_state
+            .relays
+            .iter()
+            .map(|(id, relay)| {
+                let relay = relay.map(
+                    |relay, ip4, ip6| {
+                        relay.map(|seed| {
+                            firezone_relay::Server::new(
+                                IpStack::from((ip4, ip6)),
+                                rand::rngs::StdRng::seed_from_u64(seed),
+                                3478,
+                                49152,
+                                65535,
```

Comment:
nit: It would be cool if the lowest and highest port were a Range, at first I thought this was related to the RNG seed